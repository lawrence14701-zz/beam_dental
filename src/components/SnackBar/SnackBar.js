import React from "react"
import Button from "@material-ui/core/Button"
import Snackbar from "@material-ui/core/Snackbar"
import IconButton from "@material-ui/core/IconButton"
import CloseIcon from "@material-ui/icons/Close"

export default function SimpleSnackbar(props) {
  const { isOpen, text, type } = props
  const [open, setOpen] = React.useState(isOpen)

  const handleClose = () => {
    setOpen(false)
  }

  return (
    <div>
      <Snackbar
        anchorOrigin={{
          vertical: "bottom",
          horizontal: "left",
        }}
        open={open}
        autoHideDuration={6000}
        onClose={handleClose}
        message={text}
        action={
          <React.Fragment>
            <Button color={type} size="small" onClick={handleClose}>
              {type}
            </Button>
            <IconButton
              size="small"
              aria-label="close"
              color="inherit"
              onClick={handleClose}
            >
              <CloseIcon fontSize="small" />
            </IconButton>
          </React.Fragment>
        }
      />
    </div>
  )
}
