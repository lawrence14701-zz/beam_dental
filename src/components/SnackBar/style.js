import { makeStyles } from "@material-ui/core/styles"

const useStyles = (theme) => ({
  root: {
    width: "100%",
    "& > * + *": {
      marginTop: theme.spacing(2),
    },
  },
})

export default makeStyles(useStyles)
