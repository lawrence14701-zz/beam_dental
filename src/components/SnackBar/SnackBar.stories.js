import React from "react"
import SnackBar from "./SnackBar"
import { ThemeProvider } from "@material-ui/styles"
import muiTheme from "../../Theme/theme"

export default {
  title: "SnackBar",
  component: SnackBar,
  decorators: [
    (Story) => (
      <ThemeProvider theme={muiTheme}>
        <Story />
      </ThemeProvider>
    ),
  ],
}

const Template = (args) => <SnackBar {...args} />
export const openSuccess = Template.bind({})
export const openFail = Template.bind({})
export const closed = Template.bind({})

openSuccess.args = {
  isOpen: true,
  type: "primary",
  text: "success",
}

openFail.args = {
  isOpen: true,
  text: "fail",
  type: "secondary",
}

closed.args = {
  isOpen: false,
  type: "primary",
  text: "closed",
}
