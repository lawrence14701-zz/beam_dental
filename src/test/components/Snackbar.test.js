import SnackBar from "../../components/SnackBar/SnackBar"
import "../jestsetup"
import React from "react"

function createTestProps() {
  return {
    isOpen: true,
    type: "primary",
    text: "success",
  }
}

test("renders a snackbar component", () => {
  const props = createTestProps()
  const wrapper = shallow(<SnackBar {...props} />)
  expect(wrapper).toMatchSnapshot()
})
