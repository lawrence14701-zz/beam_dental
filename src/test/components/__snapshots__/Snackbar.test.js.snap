// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snackbar component 1`] = `ShallowWrapper {}`;
